product_docs/docs/pgd/5/rel_notes/pgd_5.4.1_rel_notes.mdx

```diff
@@ -14,8 +14,9 @@ We recommend that all users of PGD 5 upgrade to PGD 5.4.1. See [PGD/TPA upgrades
 
 ## Bug fixes
 
-| Component | Version | Description                                                                                                                                                                                                                                                                                                     | Addresses        |
-|-----------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|------------------|
-| BDR       | 5.4.1  | Fixed WAL retention logic: prevent a PGD node from running out of disk space due to a bug in 5.4.0 in combination with Postgres 16, PGE 16, and EPAS 16.<br/><br/>We now make sure WAL gets properly cleaned even after 4 GB of WAL produced on a node. A change in 5.4.0 caused WAL to be retained forever after that point. This issue affects only release PGD 5.4.0 in combination with Postgres 16, PGE 16, and EPAS 16.                                                |                  |
+| Component | Version | Description                                                                                                                                                                                                                                                                                                                                                                                                                   | Addresses |
+|-----------|---------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-----------|
+| BDR       | 5.4.1   | Fixed WAL retention logic: prevent a PGD node from running out of disk space due to a bug in 5.4.0 in combination with Postgres 16, PGE 16, and EPAS 16.<hr/>We now make sure WAL gets properly cleaned even after 4 GB of WAL produced on a node. A change in 5.4.0 caused WAL to be retained forever after that point. This issue affects only release PGD 5.4.0 in combination with Postgres 16, PGE 16, and EPAS 16. |           |
```

Comment:
Fixed WAL retention logic to prevent disk space issues on PGD node with version 5.4.0 and Postgres 16, PGE 16, and EPAS 16.
We have implemented a fix to ensure proper cleanup of Write-Ahead Logs (WAL) even after reaching a size of 4 GB on a node. A change in version 5.4.0 resulted in WAL being retained indefinitely after reaching this threshold. This issue is specific to PGD 5.4.0 in conjunction with Postgres 16, PGE 16, and EPAS 16.